#
# old_revision [f81afb1976fd66fa2ad20e3700d8244c1ae29cbc]
#
# patch "mtn-browse"
#  from [560256ded1bf9b190ce9949556b71655fe6c5b5a]
#    to [62216e1a094218654e6fd005d9cb271ffbfff9d0]
#
============================================================
--- mtn-browse	560256ded1bf9b190ce9949556b71655fe6c5b5a
+++ mtn-browse	62216e1a094218654e6fd005d9cb271ffbfff9d0
@@ -79,7 +79,7 @@ use Gtk2::Pango;
 use Gtk2::GladeXML;
 use Gtk2::Helper;
 use Gtk2::Pango;
-use Gtk2::SourceView;
+use Gtk2::SourceView2;
 use IO::Dir;
 use IO::File;
 use IPC::Open3;
@@ -2405,13 +2405,13 @@ sub get_browser_window(;$$$$$)
 
         # Setup the file file viewer (with syntax highlighting).
 
-        $browser->{file_view_svbuffer} = Gtk2::SourceView::Buffer->new(undef);
-        $browser->{file_view_svbuffer}->set_check_brackets(FALSE);
+        $browser->{file_view_svbuffer} = Gtk2::SourceView2::Buffer->new(undef);
+        $browser->{file_view_svbuffer}->set_highlight_matching_brackets(FALSE);
         $browser->{file_view_svbuffer}->set_max_undo_levels(0);
         $browser->{file_view_svbuffer}->begin_not_undoable_action();
         $browser->{file_view_svlangmgr} =
-            Gtk2::SourceView::LanguagesManager->new();
-        $browser->{file_view_sv} = Gtk2::SourceView::View->
+            Gtk2::SourceView2::LanguageManager->new();
+        $browser->{file_view_sv} = Gtk2::SourceView2::View->
             new_with_buffer($browser->{file_view_svbuffer});
         $browser->{file_view_sv}->modify_font($mono_font);
         $browser->{file_view_sv}->set_cursor_visible(FALSE);
@@ -3508,7 +3508,7 @@ sub update_browser_state($$)
             $browser->{file_view_svbuffer}->
                 place_cursor($browser->{file_view_svbuffer}->get_start_iter());
             $browser->{file_view_svbuffer}->set_text("");
-            $browser->{file_view_svbuffer}->set("highlight", FALSE);
+            $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
             set_label_value($browser->{file_name_value_label}, "");
             set_label_value($browser->{file_author_value_label}, "");
             set_label_value($browser->{last_update_value_label}, "");
@@ -3634,7 +3634,7 @@ sub display_file($$)
     $browser->{file_view_svbuffer}->
         place_cursor($browser->{file_view_svbuffer}->get_start_iter());
     $browser->{file_view_svbuffer}->set_text("");
-    $browser->{file_view_svbuffer}->set("highlight", FALSE);
+    $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
 
     # Get contents.
 
@@ -3738,9 +3738,12 @@ sub display_file($$)
                          && $mime_details->{syntax_highlight}))
                     && defined($lang =
                                $browser->{file_view_svlangmgr}->
-                                   get_language_from_mime_type($mime_type)))
+                                   guess_language($browser->
+                                                      {file_being_viewed}->
+                                                      {short_name},
+                                                  $mime_type)))
                 {
-                    $browser->{file_view_svbuffer}->set("highlight", TRUE);
+                    $browser->{file_view_svbuffer}->set_highlight_syntax(TRUE);
                     $browser->{file_view_svbuffer}->set_language($lang);
                 }
 
